### PR TITLE
[NO-NSTK] Add PSO 5 to 6 Upgrade docs

### PR DESCRIPTION
**Pull request by sdodsley**
Add docs for upgrade process and link from main README.
Clean main README in existing upgrade section

### CODE REVIEW

File: docs/csi-5to6-upgradepath.md [hunk 1]
```diff
@@ -1,24 +1,48 @@
...
+
+## Upgrade Instructions
+1. :warning: **Uninstall PSO 5 using the command `helm delete -n <pso namespace> pure-storage-driver`**. 
+   * :fire: Do not delete the current k8s namespace used by PSO 5. This is required by the PSO 6 installation.
+2. Prepare your PSO 6 `values.yaml` file using values from your current PSO 5.x `values.yaml`.
+   * These files are formatted differently, so you **cannot** copy or reuse the existing PSO 5.x `values.yaml`.
+   * Schema validation is in-place during a PSO 6 installation to ensure your new file complies with the required PSO 6 format.
+3. The `upgrade` parameter in a PSO 6 `values.yaml` file is set to `false` by default.
+   * PSO will not start an upgrade process if the field is omitted or set to `false`. Set the `upgrade` parameter  to `true` to enable the upgrade process. The upgrade will takes place during the installation of PSO 6.
```
Comment (david-enli) [inline docs/csi-5to6-upgradepath.md:32]: will take maybe?

File: docs/csi-5to6-upgradepath.md [hunk 2]
```diff
@@ -1,24 +1,48 @@
...
+5. Install PSO 6 following the instruction provided [here](../pure-pso/README.md#installation).
+   * :bulb: When running the  `helm repo add` command to add the PSO 6 repository, use a different name for the chart repository to that used for the PSO 5 chart repository.
+   * :warning: You **MUST** install PSO 6 in the same k8s namespace as the original PSO 5 installation.
+   * Monitor the new PSO pods and when all pods in the PSO k8s namespace are up and running the upgrade was successful.
+
+## Rollback to PSO 5.x
+1. :warning: If an upgrade is requested in the PSO 6 `values.yaml` but the upgrade fails, PSO 6 will not start successfully.
+2. If PSO 6 is not successfully running after **5 minutes**, it is possible the upgrade failed.
+   * :information_source: Other factors, such as known CNI bugs in certain k8s versions and lack of synchronized time sources for all k8s nodes, may also attribute to installation failures. To confirm the upgrade status, enable debug mode for PSO 6 in `values.yaml`.
+   * Check the sidecar logs for `pso-csi-container` in the `pso-csi-controller-0` pod. If you need additional help, please reach out to the Pure Storage Support team.
```
Comment (david-enli) [inline docs/csi-5to6-upgradepath.md:43]: do we mean sidecar logs or container logs

**Comment by david-enli** [comment]
thanks, looks much better. just left a few comments on some small things

File: docs/csi-5to6-upgradepath.md [hunk 3]
```diff
@@ -1,24 +1,48 @@
...
-run ```kubectl get csidriver``` to see a list of csi drivers installed on the cluster. 
+1. The upgrade process will **NOT** migrate snapshots or snapshots content from PSO 5.x to PSO 6.
+   * This restriction is enforced due to PSO 6 using the beta version of K8s snapshot API, whereas PSO 5.x used the alpha version of the API with backwards compatibility not being guranteed by the CSI standard. That means that a snapshot created in PSO 5.x cannot be used to restore a volume in PSO 6 after the upgrade.
+2. :warning: When installed PSO 6 it **MUST** use the same Pure namespace as PSO 5.x. 
+   * The `namespace` -> `pure` field in the PSO 5.x helm chart `values.yaml` must be **EXACTLY** the same as the `clusterID` field in PSO 6 `values.yaml` `namespace` field in PSO 5.x has been renamed to `clusterID` in PSO 6. (These parameters are used to prefix in underlying FlashArray volumes and FlashBlade shares used by PSO).
+   * :warning: Certain restrictions were imposed on PSO 6 `clusterID` and if your PSO 5.x `namespace` setting does not meet these restrictions, this upgrade tool will not work. Please reach out to Pure Storage Support for alternative options.
```
Comment (dsupure) [inline docs/csi-5to6-upgradepath.md:22]: Can we add a link to the clusterID restrictions?

File: docs/csi-5to6-upgradepath.md [hunk 4]
```diff
@@ -1,24 +1,48 @@
...
+3. :information_source: This upgrade tool has only been scale tested to 500 PVCs/attachments. If your cluster has more than this it is possible the time taken to construct the new PSO 6 database will exceed the restart threshold of the DB controller pods. Please contact Pure Storage Support if you experience into this situation.
```
Comment (dsupure) [inline docs/csi-5to6-upgradepath.md:23]: Actually En's test passed 2400 PVCs. Can we change it to 1000?

Comment (david-enli) [reply docs/csi-5to6-upgradepath.md:23]: @dsupure 500 on a single cluster, 2400 was actually across multiple tests

Comment (david-enli) [reply docs/csi-5to6-upgradepath.md:23]: although by increasing the timeout threshold of the controller pod, I can't think of reason we upgrade more than 500 PVCs/attachments.

File: docs/csi-5to6-upgradepath.md (hunk 3 above)
Comment (sdodsley) [reply docs/csi-5to6-upgradepath.md]: Not really documented anywhere, so I detailed them here now.

File: docs/csi-5to6-upgradepath.md (hunk 4 above)
Comment (sdodsley) [reply docs/csi-5to6-upgradepath.md:23]: Increasing the timeout threshold would only be something done on a 'by request' from a customer if they have more than 500 volumes to migrate.
I'm going to leave this at 500.

**Review by david-enli — APPROVED** [review]
LGTM, should we add a note saying, if your current PSO 5 installation among multiple clusters are sharing the same namespace->pure field, then you cannot safely upgrade to PSO 6 unless there is one backend for each cluster?

**Pull request merged**